src/main/java/com/github/enimaloc/irc/jircd/internal/commands/messages/NoticeCommand.java

```diff
@@ -0,0 +1,93 @@
+import java.util.Optional;
+import java.util.function.Predicate;
+
+@Command(name = "notice")
+public class NoticeCommand {
+
+    @Command
+    public void execute(User user, String targets, String message) {
+        executeTrailing(user, targets, message);
+    }
+
+    @Command(trailing = true)
+    public void executeTrailing(User user, String targets, String trailing) {
+        for (String target : (targets + ",").split(",")) {
+            if (target.isEmpty()) {
+                continue;
+            }
+            if (Regex.CHANNEL.matcher(target.replaceFirst(Regex.CHANNEL_PREFIX.pattern(), "")).matches()) {
+                executeChannel(user, target, trailing);
+            } else if (Regex.NICKNAME.matcher(target).matches()) {
+                executeUser(user, target, trailing);
+            }
+        }
+    }
+
+    private void executeChannel(User user, String target, String trailing) {
+        String targetWithoutPrefix = target.replaceFirst(Regex.CHANNEL_PREFIX.pattern(), "");
+        String prefix              = target.replace(targetWithoutPrefix, "");
+        Optional<Channel> channelOpt = user.server()
+                                           .channels()
+                                           .stream()
+                                           .filter(c -> c.name().equals(targetWithoutPrefix))
+                                           .findFirst();
+        if (channelOpt.isEmpty()) {
+            return;
+        }
+        Channel channel = channelOpt.get();
+        if ((channel.modes().bans().stream().anyMatch(mask -> new Mask(mask).toPattern()
+                                                                            .matcher(user.info().full())
+                                                                            .matches()) &&
+             channel.modes().except().stream().noneMatch(mask -> new Mask(mask).toPattern()
+                                                                               .matcher(user.info().full())
+                                                                               .matches())) ||
+            (channel.modes().noExternalMessage() && !channel.users().contains(user)) ||
+            (channel.modes().moderate() && !Regex.CHANNEL_PREFIX.matcher(channel.prefix(user)).matches())) {
+            return;
+        }
+        Predicate<User> filter = u -> u != user;
+        if (!prefix.isEmpty()) {
+            StringBuilder builder = new StringBuilder("[");
+            for (char c : prefix.toCharArray()) {
+                switch (c) {
+                    case '+':
+                        builder.append("+");
+                    case '%':
+                        builder.append("%");
+                    case '@':
+                        builder.append("@");
+                    case '&':
+                        builder.append("&");
+                    case '~':
+                        builder.append("~");
+                }
+            }
```

Comment:
```suggestion
            StringBuilder builder = new StringBuilder("[");
            // Set<Char> operands = Stream.of('+', '%', '@', '&', '~').collect(Collectors.toSet());
            final String operands = "+%@&~";
            for (char c : prefix.toCharArray()) {
                if (operands.contains(c)) {
                    builder.append(c);
                }
            }
```

Reply:
After reflection this method would not be suitable because the message will be sent to what has the prefix and above:
> If <target> is a channel name, it may be prefixed with one or more channel membership prefix character (@, +, etc) and the message will be delivered only to the members of that channel with the given or higher status in the channel. 

[PRIVMSG documentation](https://modern.ircdocs.horse/index.html#privmsg-message) (same for NOTICE).
Perhaps by reworking this solution we could get the expected result, like: 
- % -> %@&~
- & -> &~
- etc

Reply:
Fixed version
```suggestion
            StringBuilder builder = new StringBuilder("[");

            final String operands = "+%@&~";
            for (char c : prefix.toCharArray()) {
                if (operands.contains(c)) {
                    builder.append(operands.substring(operands.indexOf(c)));
                }
            }
```
idk if that's better than your code, I'll probably stick with yours  it documenting it a bit in order to highlight the fact it has no break instructions and by that way avoid easy bug introduction in the future

Reply:
I have add a comment to describe the switch behaviors and a unit test to test this behaviors